libraries/lib-components/PluginProvider.h

```diff
@@ -157,6 +157,13 @@ class COMPONENTS_API PluginProvider  /* not final */
 
    /*! @return true if the plug-in is still valid, otherwise false. */
    virtual bool IsPluginValid(const PluginPath & path, bool bFast) = 0;
+   /**
+    * \brief Performs plugin/module existence check, still plugin may fail to load.
+    * Implementation should avoid loading plugins during this check.
+    * \param path Internal plugin path/ID discovered via DiscoverPluginsAtPath
+    * or module path returned by FindModulePaths
+    */
+   virtual bool CheckPluginExist(const PluginPath& path) const = 0;
```

Comment:
`CheckPluginExists` might be a better name.  This commit fails to build on macOS because `AudioUnitEffectsModule` needs to define an override.